library/src/main/kotlin/io/intrepid/bleidiom/util/ReplayingUnfinalizedShare.kt

```diff
@@ -0,0 +1,153 @@
+package io.intrepid.bleidiom.util
+
+import io.reactivex.Flowable
+import io.reactivex.FlowableTransformer
+import io.reactivex.Observable
+import io.reactivex.ObservableTransformer
+import io.reactivex.Observer
+import io.reactivex.disposables.Disposable
+import io.reactivex.functions.Action
+import io.reactivex.functions.Consumer
+import org.reactivestreams.Subscriber
+import org.reactivestreams.Subscription
+
+/**
+ * A transformer which combines the `replay(1)`, `publish()`, and `refCount()`
+ * operators.
+ *
+ *
+ * Unlike traditional combinations of these operators, `ReplayingShare` caches the last emitted
+ * value from the upstream observable or flowable *only* when one or more downstream subscribers
+ * are connected. This allows expensive upstream sources to be shut down when no one is listening
+ * while also replaying the last value seen by *any* subscriber to new ones.
+ *
+ * This one differs from https://github.com/JakeWharton/RxReplayingShare in that it will no longer
+ * share a value from a finalized Rx stream, i.e. it erases the last seen cached value when
+ * the upstream finalizes.
+ */
+class ReplayingUnfinalizedShare<T> private constructor() : ObservableTransformer<T, T>, FlowableTransformer<T, T> {
+
+    override fun apply(upstream: Observable<T>): Observable<T> {
+        val lastSeen = LastSeen<T>()
+        return LastSeenObservable(
+                upstream.doFinally(lastSeen).doOnNext(lastSeen).share(),
+                lastSeen
+        )
+    }
+
+    override fun apply(upstream: Flowable<T>): Flowable<T> {
+        val lastSeen = LastSeen<T>()
+        return LastSeenFlowable(
+                upstream.doFinally(lastSeen).doOnNext(lastSeen).share(),
+                lastSeen
+        )
+    }
+
+    internal class LastSeen<T> : Consumer<T>, Action {
+        @Volatile
+        var value: T? = null
+
+        override fun accept(latest: T) {
+            value = latest
+        }
+
+        override fun run() {
```

Comment:
This `run` method is called when the upstream chain, *before* it is `share`d, finalizes. When this happens, the LastSeen's `value` is set to `null` and it will no longer be replayed until a next emission happens.